### PR TITLE
Filter the TestPackage artifact upload by name to avoid other zip files being uploaded 

### PR DESCRIPTION
**Pull request by adityapatwardhan**
## PR Summary

Added logging for `New-TestPackage` to make debugging easier.
Filter the artifact uploads with the package name instead of *.zip to avoid Help Zip files from being uploaded.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
I have verified the changes on a private CI run, which is a clone of the official Azure DevOps pipeline.

File: .vsts-ci/mac.yml [hunk 1]
```diff
@@ -53,7 +53,8 @@ phases:
 
   # Uploads any packages as an artifact
   - powershell: |
-      Get-ChildItem -Path *.pkg, *.tar.gz, *.zip -Recurse | Select-Object -ExpandProperty FullName | ForEach-Object {
+      Write-Host "PWD == $pwd"
```
Comment (anmenaga) [inline .vsts-ci/mac.yml]: As far as I remember `$pwd` variable should Not be used in scripts. Change to Get-Location.

**Review by anmenaga — APPROVED** [review]
1 update needed; otherwise looks good.

File: .vsts-ci/mac.yml (hunk 1 above)
Comment (JamesWTruher) [reply .vsts-ci/mac.yml]: it's likely in this context it's ok, but something that we should be aware of.

**Review by JamesWTruher — APPROVED** [review]
my comment isn't blocking

**Comment by adityapatwardhan** [comment]
@JamesWTruher  @anmenaga  Remove the usage of `$pwd` it was a debug statement which can now be removed.

**Comment by adityapatwardhan** [comment]
I will merge when CI passes.

**Pull request merged**